src/harness/reference_models/pre_iap_filtering/fss_purge.py

```diff
@@ -0,0 +1,301 @@
+FSS_OOBE_MARGIN = 1
+
+# Define named tuple
+grants_cbsds_namedtuple = namedtuple('grants_cbsds_namedtuple', ['grant', 'cbsd'])
+
+def getFssInfo(fss_record):
+  """ Create a FssProtectionPoint tuple from the fss_record.
+
+  Args:
+    fss_record: A single FSS record dictionary.
+  Returns:
+    FssProtectionPoint: Tuple holding details about FSS needed for interference calculation.
+  """
+
+  # Extract fields from FSS record.
+  fss_latitude = fss_record['record']['deploymentParam'][0]['installationParam']['latitude']
+  fss_longitude = fss_record['record']['deploymentParam'][0]['installationParam']['longitude']
+  fss_height = fss_record['record']['deploymentParam'][0]['installationParam']['height']
+  fss_height_type = fss_record['record']['deploymentParam'][0]['installationParam']['heightType']
+  fss_azimuth = fss_record['record']['deploymentParam'][0]['installationParam']['antennaAzimuth']
+  fss_pointing_elevation = - fss_record['record']['deploymentParam'][0]['installationParam']['antennaDowntilt']
+  fss_max_gain_dbi = fss_record['record']['deploymentParam'][0]['installationParam']['antennaGain']
+
+  # Convert the height to AGL if it was AMSL type.
+  if fss_height_type == 'AMSL':
+    fss_height_agl = convertAmslToAgl(fss_latitude, fss_longitude, fss_height)
+  elif fss_height_type == 'AGL':
+    fss_height_agl = fss_height
+  else:
+    raise Exception('Invalid FSS height type.')
+
+  # Populate FssProtectionPoint tuple.
+  fss_entity = interference.FssProtectionPoint(latitude=fss_latitude,
+                                               longitude=fss_longitude,
+                                               height_agl=fss_height_agl,
+                                               max_gain_dbi=fss_max_gain_dbi,
+                                               pointing_azimuth=fss_azimuth,
+                                               pointing_elevation=fss_pointing_elevation)
+  return fss_entity
+
+def generatePurgeListForFssPoint(cbsds, fss_entity):
+  """ Implementation of algorithm described in R2-SGN-29.
+
+  The list of grants to be purged based on the interference caused by their
+  out of band emission are identified and returned.
+  Args:
+    cbsds: List of CbsdData objects that are in the neighborhood to an FSS.
+    fss_entity: The FssProtectionPoint tuple of the FSS for which the purge is to be performed.
+  Returns:
+    List of multiple key-value pairs with keys "grantData", "cbsd" and values as grants to be
+    purged and corresponding cbsd data object respectively.
+  """
+  # List of grants to be purged for the fss_entity.
+  final_purge_list = []
+
+  # List of dictionaries containing the grant_info based on the highest grant from FSS OOBE calculation
+  grants_cbsds_info_for_oobe_calculation = getGrantInfoListForFssOobeCalculation(cbsds)
+
+  # Calculate the threshold value in dBm for the OOBE interference
+  oobe_threshold_value = -129 + 10 * (np.log10(REF_BW/interference.ONE_MHZ) - FSS_OOBE_MARGIN)
```

Comment:
The closing bracket seems incorrect. 
Proposed update:
oobe_threshold_value = -129 + 10 * (np.log10(REF_BW/interference.ONE_MHZ)**)** - FSS_OOBE_MARGIN